docs/runtime/initializing-storage.md

```diff
@@ -0,0 +1,140 @@
+---
+title: "Initializing Runtime Storage"
+---
+
+Most runtime modules contain storage values that may be changed or updated as the blockchain runs and users interact with the module. The ways that data may change are coded in the runtime module's functions (both dispatchable and not), and are covered in several [tutorials](/tutorials).
+
+In some cases storage should be initialized with a value before any user has interacted with it. This is primarily the job of the genesis configuration. There are four ways runtime developers may specify initial values for their storage items, and this article will cover all four. Another article discusses how the genesis configuration works [behind the macros](runtime/types/genesisconfig-struct.md).
+
+## Hard-Code Specific Values
+
+The simplest way (and one that technically does not use genesis configuration) to initialize storage is simply to hard-code specific values for each parameter. This option is best when there are clear default values that all deployments of the module will want to use. The runtime developer can choose this option by including `config()` in the `decl_storage!` marco, and an initial value at the end of the line.
+
+Consider this example module where players ante to play a game and add to the pot by raising.
+
+```rust
+decl_storage! {
+  trait Store for Module<T: Trait> as TemplateModule {
+    // Points ante'd to play the game
+    pub Ante get(ante) config(): u32 = 5;
+
+    // Minimum raise on each turn
+    pub MinRaise get(min_raise) config(): u32 = 7;
+  }
+}
+```
```

Comment:
Actually the Hard-Code default value is not working using config. It can be used without config at all (and without adding config to construct_runtime either)

And the behavior is quite surprising.
What does `= 5` means is that the default value of the storage is 5.
Thus the following test is valid:
```diff
diff --git a/srml/example/src/lib.rs b/srml/example/src/lib.rs
index 7e197c13e..1306c43a9 100644
--- a/srml/example/src/lib.rs
+++ b/srml/example/src/lib.rs
@@ -302,6 +302,8 @@ decl_storage! {
 
                // this one uses the default, we'll demonstrate the usage of 'mutate' API.
                Foo get(foo) config(): T::Balance;
+
+               E: Vec<u64> = vec![1, 3];
        }
 }
 
@@ -603,4 +605,15 @@ mod tests {
                        assert_eq!(Example::foo(), 25);
                });
        }
+
+       #[test]
+       fn toto() {
+               with_externalities(&mut new_test_ext(), || {
+                       assert_eq!(E::get(), vec![1, 3]);
+                       E::put(vec![]);
+                       assert_eq!(E::get(), vec![]);
+                       E::kill();
+                       assert_eq!(E::get(), vec![1, 3]);
+               });
+       }
 }
```

Reply:
Oh but you are also right I actually didn't know: if you use the hard coded default + the config  then it will implement config default this way:
```rust
#[cfg(feature = "std")]
impl <T: Trait> Default for GenesisConfig<T> {
    fn default() -> Self {
        GenesisConfig{dummy: Default::default(),
                      bar: Default::default(),
                      foo: Default::default(),
                      e: <[_]>::into_vec(box [1, 3]),}
    }
}
```

thus writing 
```rust
		t.extend(GenesisConfig::<Test>{
			.. Default::default()
		}.build_storage().unwrap().0);
```
will give the value `[1, 3]` to e
but writing:
```rust
		t.extend(GenesisConfig::<Test>{
			e: Default::default(),
		}.build_storage().unwrap().0);

```
will give the value `Vec<u64>::default` which is `vec![]`

that looks very confusing, also maybe it worth investigate if we want to implement something more straightforward than this. But this is the current behavior.

Reply:
Oh interesting! I also didn't realize all of that.

 I'm not sure it's wise to explain so much subtle detail in this article since the target audience is someone building their first few modules. I've added a callout noting that if you kill the storage value, it will return to the default.

Reply:
it has this effect if you kill or if you `take` it.